bfcfg

```diff
@@ -1171,33 +1176,91 @@ sb_cfg()
   fi
 }
 
-usage()
+# Convert input configuration to a binary file. As of
+# now only MFG configuration is converted.
+#
+# Note 1: MFG dependencies such as system attributes
+#         and Redfish settings are not supported.
+#
+# Note 2: This function can be enhanced to support
+#         additional configuration.
+#
+# When 'bfcfg' cannot be executed from Arm operating
+# system, an alternative is to run the command on external
+# host to convert the config to a binary file. This binary
+# file can be used to create a capsule that is appended to
+# a BFB file and applied from external host.
+cfg2bin()
+{
+  local cfg_file=$1
+  local bin_file=${cfg_file}.bin
+  local tmp_bin_file=/tmp/.bfcfg-mfg.bin
+
+  # Source the configuration if exists.
+  if [ -e "${cfg_file}" ]; then
+    . ${cfg_file}
+  else
+    echo "error: missing configuration file."
+    return
+  fi
+
+  # Reset variables
+  mfg_sysfs_cfg_done=0
+  has_redfish_cfg=0
+  has_sys_cfg=0
+
+  start_log_file
+
+  # Export config to bin
+  mfg_cfg_to_bin ${tmp_bin_file}
+  has_bin=0
+
+  [ -f "${tmp_bin_file}" ] && has_bin=1
+
+  if [ ${has_bin} -eq 1 ]; then
+    mv ${tmp_bin_file} ${bin_file}
+    rm -f ${tmp_bin_file}
+    echo ""
+    echo "$0: configuration blob written to ${bin_file}"
+  else
+    echo "error: failed to write configuration."
+    return
+  fi
+}
+
+start_log_file()
 {
-  echo "syntax: bfcfg [--help|-h] [--dump|-d] [--part-info|-p]"
+  rm -f ${log_file} >/dev/null
+  log_msg "bfcfg (ver ${bfcfg_version})"
+  log_msg "$(date)"
+  log_msg
 }
 
-# Source the configuration if exists.
-# shellcheck source=/dev/null
-[ -e "${cfg_file}" ] && . ${cfg_file}
+usage()
+{
+  echo "syntax: bfcfg [--help|-h] [--dump|-d] [--part-info|-p] [--cfg2bin|-c CFGFILE]"
+}
 
 # Parse the arguments.
-options=$(getopt -n bfcfg -o dhp -l dump,help,part-info -- "$@")
+options=$(getopt -n bfcfg -o d,h,p,c: -l dump,help,part-info,cfg2bin: -- "$@")
```

Comment:
This change doesn't look correct to me. It adds ',' as as potential argument. Or do you see any official document to prefer the 'd,h,p' formay other than 'dhp'?

Reply:
A similar usage in bfsbverify might need fix as well

Reply:
For some reason, I thought the comma is okay. I'll fix it.

What do you think we should do about bfsbverify? send in a separate commit?

Reply:
I posted a second commit to address the issue in bfsbverify command